src/features/creator-request/server.ts

```
@@ -0,0 +1,145 @@
+import { createServerFn } from "@tanstack/react-start";
+import { eq } from "drizzle-orm";
+import {
+  approveCreatorRequestContract,
+  rejectCreatorRequestContract,
+  submitCreatorRequestContract,
+} from "@/contracts/creator-request.contract";
+import { db } from "@/db";
+import { creatorRequest, user } from "@/db/schema";
+import { getAuthSession } from "@/lib/auth-actions";
+import { requireAdmin, requireMember } from "@/lib/auth-middleware";
+
+export const submitRequest = createServerFn({ method: "POST" })
+  .middleware([requireMember])
+  .inputValidator(submitCreatorRequestContract)
+  .handler(async ({ data, context }) => {
+    const newRequest = {
+      userId: context.user.id,
+      portfolioLink: data.portfolioLink,
+      motivation: data.motivation,
+    };
+
+    const result = await db
+      .insert(creatorRequest)
+      .values(newRequest)
+      .returning({ id: creatorRequest.id });
+
+    return result[0];
+  });
+
+export const approveRequest = createServerFn({ method: "POST" })
+  .middleware([requireAdmin])
+  .inputValidator(approveCreatorRequestContract)
+  .handler(async ({ data, context }) => {
+    const { userId, requestId, note } = data;
+    try {
+      await db.update(user).set({ role: "creator" }).where(eq(user.id, userId)).returning();
+      await db
+        .update(creatorRequest)
+        .set({ status: "approved", reviewedBy: context.user.id, note })
+        .where(eq(creatorRequest.id, requestId));
+    } catch (error) {
+      if (error instanceof Error) console.log(error.message);
+    }
```

I guess it's on me for just saying list functions on https://github.com/Minilemon-Official-Team/PGT-4-Pelari-Kalcer/pull/31#discussion_r2609078461 instead of any functions. But please change this too, since it also swallows the errors and returns `undefined` to the client.

Suggestion: remove the try/catch block in the server functions and let the error propagate to the client, or return a structured result object (e.g., { success: false, error: ... }) that the client can handle explicitly.